app/src/main/java/com/android/sample/ui/listActivities/LikedActivities.kt

```diff
@@ -0,0 +1,254 @@
+import com.android.sample.model.activity.ListActivitiesViewModel
+import com.android.sample.model.profile.ProfileViewModel
+import com.android.sample.model.profile.User
+import com.android.sample.ui.navigation.BottomNavigationMenu
+import com.android.sample.ui.navigation.LIST_TOP_LEVEL_DESTINATION
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Screen
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter", "SuspiciousIndentation")
+@Composable
+fun LikedActivitiesScreen(
+    viewModel: ListActivitiesViewModel,
+    navigationActions: NavigationActions,
+    profileViewModel: ProfileViewModel,
+    modifier: Modifier = Modifier
+) {
+  val uiState by viewModel.uiState.collectAsState()
+  val profile = profileViewModel.userState.collectAsState().value
+  val allActivities = (uiState as ListActivitiesViewModel.ActivitiesUiState.Success).activities
+
+  Scaffold(
+      modifier = modifier.testTag("likedActivitiesScreen"),
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      }) { paddingValues ->
+        Box(modifier = modifier.fillMaxSize().padding(paddingValues)) {
+          val likedActivitiesList by remember { mutableStateOf(profile?.likedActivities) }
+          when (uiState) {
+            is ListActivitiesViewModel.ActivitiesUiState.Success -> {
+              if (profile == null) {
+                Text(
+                    text = "You are not logged in, Login or Register to see your liked activities",
+                    modifier =
+                        Modifier.padding(8.dp)
+                            .align(Alignment.Center)
+                            .testTag("notConnectedPrompt"),
+                    color = MaterialTheme.colorScheme.onSurface)
+                Button(
+                    onClick = { navigationActions.navigateTo(Screen.SIGN_UP) },
+                    modifier = Modifier.testTag("signInButton")) {
+                      Text("Go to Sign In Page")
+                    }
+              }
+              if (likedActivitiesList != null) {
+                if (likedActivitiesList!!.isEmpty()) {
+                  Text(
+                      text = "There is no liked activity yet.",
+                      modifier =
+                          Modifier.padding(8.dp)
+                              .align(Alignment.Center)
+                              .testTag("emptyLikedActivityPrompt"),
+                      color = MaterialTheme.colorScheme.onSurface)
+                } else {
+
+                  LazyColumn(
+                      modifier =
+                          Modifier.padding(paddingValues)
+                              .fillMaxSize()
+                              .padding(16.dp)
```

Comment:
The handling of different UI states (Success, Error, and conditionally based on user authentication) demonstrates a clear understanding of dynamic UI updates based on state changes, which is crucial for a reactive user interface.